### PR TITLE
Create Book Appointments Front-end

### PR DESCRIPTION
**Pull request by thecodechaser**
## This branch includes the following features:
- [x] Created front-end for Book Appointments application.
- [x] Added Redux store.
- [x] Added Redux reducers to handle authentication and API data.
- [x] Created Sign up and login forms.
- [x] Created navigations.
- [x] Created components for all pages.
- [x] Added API-docs links.
- [x] Applied styles
- [x] Added responsiveness.

### Link to the backend: [Book Apponitments back-end](https://github.com/Chimwemwe-127001/book-appointment-backend)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            </PrivateRoute>
+          )}
+        />
+        <Route
+          path="/reservations"
+          element={(
+            <PrivateRoute>
+              <Reservations />
+            </PrivateRoute>
+          )}
```
Comment (mcrd25) [inline src/App.js:37]: Maybe there is a cleaner way to do this?

File: src/components/sessions/Login.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              <input type="email" id="email" ref={emailRef} className="input_field" required />
+              <label htmlFor="email" className="input_label">
+                Email address
+              </label>
+            </div>
+            <div className="form-group">
+              <input type={showPassword ? 'text' : 'password'} id="password" ref={passwordRef} className="input_field" required />
+              <label htmlFor="password" className="input_label">
+                Password
+              </label>
```
Comment (mcrd25) [inline src/components/sessions/Login.js:71]: Consider elements where that can be made into components. Try best not to repeat self with forms and Inputs or buttons. Especially when it's being used across different components. So yes Add doctor and reservation are forms but maybe form itself can be a components and input etc.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi @thecodechaser @Buyaki01 @AnselemOdims @Chimwemwe-127001 @mwanawabangona  , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there! I really like the effort you've put into trying to follow design as much as possible. I noticed also that you sort of filled the gap with some missing pieces of the design in terms of it's feel, which is great.


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- What you need to do is remove the API doc button (a user should not even be thinking about the back-end of the application their focus is using the application to accomplish something)
- While you did a great job with the UI, remember you are following a design. In real world it's about what the client wants. They've already paid someone to design it. It's your job to develop it. That being said great job, but there is still some work to do to make it closer to design (particularly the reserve page)
- There are some things I'd like you to improve concerning the UI - see loom video: https://www.loom.com/share/78e8ad6d93bf4602b1068dd80780e852

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by thecodechaser** [comment]
@mcrd25 Hi, Thank you for your time and feedback 😄

**Review by MuriloRoque — APPROVED** [review]
Hi @thecodechaser, @Buyaki01, @AnselemOdims, @Chimwemwe-127001,

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by mcrd25** [comment]
> @mcrd25 Hi, Thank you for your time and feedback 😄

Np!

If you liked my review, feel free to connect on LinkedIn and leave a skill endorsement: https://www.linkedin.com/in/mayadouglas/

You are also free to follow me on GitHub

**Pull request merged**